suite-native/module-add-accounts/src/screens/SelectAccountTypeScreen.tsx

```diff
@@ -0,0 +1,149 @@
+import { AccountType } from '@suite-common/wallet-config';
+import {
+    StackNavigationProps,
+    AddCoinAccountStackParamList,
+    AddCoinAccountStackRoutes,
+    Screen,
+    ScreenSubHeader,
+    StackProps,
+} from '@suite-native/navigation';
+import { Button, VStack, Text, IconButton } from '@suite-native/atoms';
+import { useTranslate, Translation } from '@suite-native/intl';
+import { useOpenLink } from '@suite-native/link';
+
+import { useAddCoinAccount } from '../hooks/useAddCoinAccount';
+import { SelectableAccountTypeItem } from '../components/SelectableAccountTypeItem';
+import { getIntlKeysForAccountType } from '../utils/getIntlKeysForAccountType';
+
+const GRADIENT_HEIGHT = 48;
+
+const ACCOUNT_TYPES_URL = 'https://trezor.io/learn/a/multiple-accounts-in-trezor-suite';
+
+const itemsStyle = prepareNativeStyle(utils => ({
+    paddingHorizontal: utils.spacings.extraSmall,
+}));
+
+const bottomWrapperStyle = prepareNativeStyle((_, { bottomInset }: { bottomInset: number }) => ({
+    position: 'absolute',
+    paddingBottom: bottomInset,
+    left: 0,
+    right: 0,
+    bottom: 0,
+}));
+
+const gradientStyle = prepareNativeStyle(_ => ({
+    height: GRADIENT_HEIGHT,
+}));
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    paddingHorizontal: utils.spacings.medium,
+    backgroundColor: utils.colors.backgroundSurfaceElevation0,
+}));
+
+const aboutStyle = prepareNativeStyle((utils, { bottomInset }: { bottomInset: number }) => ({
+    paddingTop: utils.spacings.extraLarge,
+    paddingBottom: 2 * GRADIENT_HEIGHT + bottomInset,
+    width: '100%',
+    paddingHorizontal: utils.spacings.small,
+    gap: 12,
+}));
+
+export const SelectAccountTypeScreen = ({
+    route,
+}: StackProps<AddCoinAccountStackParamList, AddCoinAccountStackRoutes.SelectAccountType>) => {
+    const { defaultType, network } = route.params;
+    const { translate } = useTranslate();
+    const openLink = useOpenLink();
+    const insets = useSafeAreaInsets();
+    const { applyStyle, utils } = useNativeStyles();
+    const navigation =
+        useNavigation<
+            StackNavigationProps<
+                AddCoinAccountStackParamList,
+                AddCoinAccountStackRoutes.SelectAccountType
+            >
+        >();
+
+    const { getAvailableAccountTypesForNetwork, addCoinAccount, isAddingCoinAccount } =
+        useAddCoinAccount();
+    const [selectedAccountType, setSelectedAccountType] = useState<AccountType>(defaultType);
+
+    const onPressSelectedAccountType = (type: AccountType) => setSelectedAccountType(type);
+
+    const types: AccountType[] = getAvailableAccountTypesForNetwork({ network });
+    const { titleKey: accountTypeKey } = getIntlKeysForAccountType(selectedAccountType);
+
+    const handleClose = () => navigation.goBack();
+
+    const handleMoreTap = () => openLink(ACCOUNT_TYPES_URL);
+
+    const handleConfirmTap = () => {
+        if (!isAddingCoinAccount) {
```

Comment:
Isn't it better to make the button disabled so the click doesn't happen when `isAddingCoinAccount` is fulfilled?

Reply:
I'll probably remove this completely since the flow will work differently after I received design from Kuba for loading

Reply:
5c5f2af2c4